l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -259,6 +257,13 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Initializable, Owna
         emit BridgehubDepositFinalized(_chainId, _txDataHash, _txHash);
     }
 
+    /// @dev Sets the L1ERC20Bridge contract address. Should be called only once.
+    function setL1Erc20Bridge(address _legacyBridge) external onlyOwner {
+        require(address(legacyBridge) == address(0), "ShB: legacy bridge already set");
+        require(_legacyBridge != address(0), "ShB: legacy bridge 0");
+        legacyBridge = IL1ERC20Bridge(_legacyBridge);
+    }
```

Comment:
@kelemeno is it safe to have an uninitialized value?
Should we have some kind of protection?

Reply:
Had a chat with @kelemeno we considered this solution, as an optimal for now